readline/src/main/java/org/aesh/readline/terminal/TerminalBuilder.java

```diff
@@ -156,4 +158,19 @@ private Terminal createWindowsTerminal(String name) throws IOException {
                     (out == null) ? System.out : out);
         }
     }
+
+    // Console.isTerminal() was introduced in Java 22
+    private static boolean isTerminal(Console console) {
+        try {
+            Method isTerminal = Console.class.getMethod("isTerminal");
```

Comment:
Not sure if it's worth switching to a multi-versioned jar instead of relying on reflection + exception?

Reply:
cc @stalep

Reply:
Hm, instead of reflection could we check if `Runtime.version() > 21` ?
Tbh, I'm not sure what's the "best" solution here, I'm open for anything. This method is usually not called more than once anyway...

Reply:
Then the new method isTerminal won't compile on older JDKs (<= 21).

Reply:
@gsmet @stalep Submitted an alternative PR that uses "multi-release jar" feature: #70